main/test/ca/uwaterloo/flix/language/phase/TestParser.scala

```diff
@@ -115,6 +127,7 @@ class TestParser extends AnyFunSuite with TestUtils {
     expectError[ParseError](result)
   }
 
+  // TODO: move to Lexer
   test("ParseError.Int32.01") {
     val input =
```

Comment:
Not everything has been annotated, but I agree with moving all the `ParseError.IntXYZ` tests to `TestLexer`.

Reply:
I would drop the regression test.

Reply:
As for the EOI, I guess they belong here?

Reply:
Darn, I lost some comments from trying to work in multiple editors at the same time. (file saved in one, the other recovers the file to it's previous version.)

I'll redo and push.

Reply:
There. Sorry about the inconvenience!

Reply:
TLDR: 
Move all the number literal tests to lexer
Move all the EOI tests to Lexer. They are unclosed strings and chars.
Delete regression tests
Keep the rest

Reply:
I agree. Please do it :)